assets/js/util/tracking/index.js

```diff
@@ -0,0 +1,194 @@
+/**
+ * Event tracking utilities.
+ *
+ * Site Kit by Google, Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+const DEFAULT_CONFIG = {
+	isFirstAdmin: false,
+	trackingEnabled: false,
+	trackingID: '',
+	referenceSiteURL: '',
+	userIDHash: '',
+};
+
+/**
+ * Tracking configuration.
+ */
+const config = { ...DEFAULT_CONFIG };
+
+/**
+ * The promise created when tracking is enabled.
+ * This reference is kept and returned for subsequent calls.
+ */
+let enableTrackingPromise;
+
+/**
+ * gtag script identifier.
+ */
+const SCRIPT_IDENTIFIER = 'data-googlesitekit-gtag';
+
+/**
+ * Data layer global used for internal/private Site Kit data.
+ */
+export const DATA_LAYER = '_googlesitekitDataLayer';
+
+/**
+ * Initializes tracking.
+ *
+ * @param {Object} _global The global object. Optional. (Used for testing only)
+ * @param {boolean} _resetPromise Clears the internal reference to the enableTrackingPromise (Optional - for testing only)
+ */
+export async function bootstrapTracking( _global = global, _resetPromise = false ) {
```

Comment:
If we need flexibility here, I'd much rather have us pass `initialConfig` as an object and `dataLayer` as an array. The latter could be stored for future access. Let's be specific rather than passing the entire `_global`.